packages/react-form-state/src/FormState.ts

```diff
@@ -230,9 +244,9 @@ export default class FormState<
 
     if (errors.length > 0) {
       this.updateRemoteErrors(errors);
-      this.setState({submitting: false});
+      this.setState({submitting: false, lastSubmitErrors: []});
     } else {
-      this.setState({submitting: false, errors});
+      this.setState({submitting: false, lastSubmitErrors: [], errors});
```

Comment:
Alternatively, we could just also pass `errors` to `lastSubmitErrors` here, but that might end up being confusing / redundant.